zerver/lib/outgoing_webhook.py

```diff
@@ -8,17 +8,45 @@
 import logging
 from six.moves import urllib
 from functools import reduce
+from requests import Response
 
 from django.utils.translation import ugettext as _
 
-from zerver.models import Realm, get_realm_by_email_domain, get_user_profile_by_id, get_client
+from zerver.models import Realm, UserProfile, get_realm_by_email_domain, get_user_profile_by_id, get_client
 from zerver.lib.actions import check_send_message
 from zerver.lib.queue import queue_json_publish
 from zerver.lib.validator import check_dict, check_string
 from zerver.decorator import JsonableError
 
 MAX_REQUEST_RETRIES = 3
 
+class OutgoingWebhookServiceInterface(object):
+
+    def __init__(self, base_url, token, user_profile, service_name):
+        # type: (Text, Text, UserProfile, Text) -> None
+        self.base_url = None # type: Text
```

Comment:
Yeah I wanted to, probably forgot about it in the way. I am not sure about the necessity here, because it is not doing anything special but at the same time inheriting classes wouldn't be required to overwrite it. So if you think it would be better to remove it, it would be fine for me.